### ISSUE TITLE
Wallet file encryption

### ISSUE THREAD

**Issue by gak** [feeless/feeless#14]
It would be good to optionally have the wallet encrypted. For now I think the whole wallet file should be encrypted, rather than individual wallets. A new issue should be made for individual wallets.

This should use a well established cryptography algorithm.

An example to encrypt the whole wallet file:
```bash
export FEELESS_WALLET_FILE=gak.wallet

feeless wallet password
Enter a new password:
Confirm password:
Password added.
```

Doing it again will ask for the existing password:
```bash
feeless wallet password --file gak.wallet
Enter existing password:
Enter a new password:
Confirm password:
Password replaced.
```

When accessing keys, a password will be requested:
```bash
feeless wallet address --file gak.wallet
Enter password:
nano_1abcd...
```

You can remove a password and make it unencrypted again:
```bash
feeless wallet password --remove
Enter existing password:
Password removed.
```

**Comment by Fiono11** [comment]
For encryption maybe we could use one of these? https://github.com/RustCrypto/stream-ciphers

**Comment by gak** [comment]
Yes, I think so. Not sure what's good though. Maybe see what other crypto wallets (Nano, Ethereum, Bitcoin) are using?

**Comment by gak** [comment]
Bitcoin seems to use AES/SHA512

https://github.com/bitcoin/bitcoin/blob/c7ad94428ab6f54661d7a5441e1fdd0ebf034903/src/wallet/crypter.cpp#L46

**Comment by gak** [comment]
Did some more research. Let's go with https://crates.io/crates/ctr

**Comment by Fiono11** [comment]
Working on this issue.
